### PR TITLE
Contrib section

### LINKED ISSUE #1057: Make the Contributing Module a Section
Right now, we have a [module on contributing](https://usaco.guide/general/contributing/) and some [content docs](https://github.com/cpinitiative/usaco-guide/blob/master/docs/Content%20Documentation.md).

Would it be better if we made a new group of modules for Contributing and eliminated most of the content docs files?
(the benefits are that it looks nicer, can be edited with the editor, can contain MDX, and will be more accessible)

The section would tentatively contain:
- Getting Started (ways to contribute, dev server & editor)
- Editing Modules (how modules work, how to edit them, some basic and common components, philosophy of modules and when to create a new module)
- Adding Solutions (basically [this section](http://localhost:8000/general/contributing/#adding-a-solution))
- Working with MDX (exhaustive list of all of the different components, some internals on how MDX works?)
- Front End Development (working with the js code for the website)

### PR DESCRIPTION
**Pull request by bqi343**
closes #1057

### CODE REVIEW

File: content/1_General/Working_MDX.mdx [hunk 1]
```diff
@@ -375,19 +326,11 @@ author: Nathan Wang
 ```
 
 The ID of the solution frontmatter must be the same as the unique ID of the
-problem. Make sure to also update the problem metadata to "internal" for any
+problem. Make sure to also update the problem metadata to `'internal'` for any
```
Comment (maggie-j-liu) [inline content/1_General/Working_MDX.mdx:329]:
```suggestion
problem. Make sure to also update the `kind` of the `solutionMetadata` to `'internal'` for any
```
maybe this is more accurate?

File: content/1_General/Working_MDX.mdx [hunk 2]
```diff
@@ -375,19 +326,11 @@ author: Nathan Wang
...
-
-Example usage:
-
-```mdx
-<Problems problems="problems" />
-```
-
-`[module].problems.json` should have a key of `problems` that maps to an array
-of `ProblemMetadata`.
+solution but the problem metadata isn't set to `'internal'`. The
```
Comment (maggie-j-liu) [inline content/1_General/Working_MDX.mdx:332]:
```suggestion
solution but the `solutionMetadata`'s `kind` isn't set to `'internal'`. The
```

File: content/1_General/Working_MDX.mdx [hunk 3]
```diff
@@ -1,68 +1,19 @@
...
+  'Explanation of the frontmatter that precedes every module and solution, a
+  list of custom components that may be used within modules or solutions.'
+prerequisites:
+  - modules
+---
 
-Located at `content/ordering.ts`, this file stores the ordering of the modules.
-Hopefully the format is self-explanatory (it matches based on "slug"). Let
-Nathan Wang know if you have questions.
+We're using [MDX](https://mdxjs.com/), a superset of Markdown. HTML and React
```
Comment (maggie-j-liu) [inline content/1_General/Working_MDX.mdx:12]: maybe mention somewhere that the compiler is xdm so `<` and `>` need to be escaped

**Review by thecodingwizard — APPROVED** [review]
Seems good enough for now

**Pull request merged**